packages/opencensus-core/src/metrics/cumulative/cumulative.ts

```diff
@@ -0,0 +1,196 @@
+    const keysAndConstantKeys = [...labelKeys, ...constantLabels.keys()];
+    this.constantLabelValues = [...constantLabels.values()];
+
+    this.metricDescriptor =
+        {name, description, unit, type, labelKeys: keysAndConstantKeys};
+    this.defaultLabelValues = initializeDefaultLabels(this.labelKeysLength);
+  }
+
+  /**
+   * Creates a TimeSeries and returns a Point if the specified
+   * labelValues is not already associated with this cumulative, else returns an
+   * existing Point.
+   *
+   * It is recommended to keep a reference to the Point instead of always
+   * calling this method for manual operations.
+   *
+   * @param {LabelValue[]} labelValues The list of the label values.
+   * @returns {CumulativePoint} The value of single cumulative.
+   */
+  getOrCreateTimeSeries(labelValues: LabelValue[]): CumulativePoint {
+    validateArrayElementsNotNull(
+        validateNotNull(labelValues, 'labelValues'), 'labelValue');
+    return this.registerTimeSeries(labelValues);
+  }
+
+  /**
+   * Returns a Point for a cumulative with all labels not set, or default
+   * labels.
+   *
+   * @returns {CumulativePoint} The value of single cumulative.
+   */
+  getDefaultTimeSeries(): CumulativePoint {
+    return this.registerTimeSeries(this.defaultLabelValues);
+  }
+
+  /**
+   * Removes the TimeSeries from the cumulative metric, if it is present. i.e.
+   * references to previous Point objects are invalid (not part of the
+   * metric).
+   *
+   * @param {LabelValue[]} labelValues The list of label values.
+   */
+  removeTimeSeries(labelValues: LabelValue[]): void {
+    validateNotNull(labelValues, 'labelValues');
+    this.registeredPoints.delete(hashLabelValues(labelValues));
+  }
+
+  /**
+   * Removes all TimeSeries from the cumulative metric. i.e. references to all
+   * previous Point objects are invalid (not part of the metric).
+   */
+  clear(): void {
+    this.registeredPoints.clear();
+  }
+
+  /**
+   * Registers a TimeSeries and returns a Point if the specified
+   * labelValues is not already associated with this cumulative, else returns an
+   * existing Point.
+   *
+   * @param {LabelValue[]} labelValues The list of the label values.
+   * @returns {CumulativePoint} The value of single cumulative.
+   */
+  private registerTimeSeries(labelValues: LabelValue[]): CumulativePoint {
+    const hash = hashLabelValues(labelValues);
+    // return if the specified labelValues is already associated with the point.
+    if (this.registeredPoints.has(hash)) {
+      return this.registeredPoints.get(hash)!;
+    }
+    if (this.labelKeysLength !== labelValues.length) {
+      throw new Error('Label Keys and Label Values don\'t have same size');
+    }
+
+    const point =
+        new CumulativePointEntry([...labelValues, ...this.constantLabelValues]);
+    this.registeredPoints.set(hash, point);
+    return point;
+  }
+
+  /**
+   * Provides a Metric with one or more TimeSeries.
```

Comment:
Can you update this comment? And also explain when it would return `null`?

Reply:
Done, basically `null` is returned when TimeSeries is not present in Metric.